Source/Core/Core/Analytics.cpp

```diff
@@ -237,21 +237,14 @@ void DolphinAnalytics::MakeBaseBuilder()
 #if defined(_WIN32)
   builder.AddData("os-type", "windows");
 
-  // Windows 8 removes support for GetVersionEx and such. Stupid.
-  DWORD(WINAPI * RtlGetVersion)(LPOSVERSIONINFOEXW);
-  *(FARPROC*)&RtlGetVersion = GetProcAddress(GetModuleHandle(TEXT("ntdll")), "RtlGetVersion");
-
-  OSVERSIONINFOEXW winver;
+  OSVERSIONINFOEX winver;
   winver.dwOSVersionInfoSize = sizeof(winver);
-  if (RtlGetVersion != nullptr)
-  {
-    RtlGetVersion(&winver);
-    builder.AddData("win-ver-major", static_cast<u32>(winver.dwMajorVersion));
-    builder.AddData("win-ver-minor", static_cast<u32>(winver.dwMinorVersion));
-    builder.AddData("win-ver-build", static_cast<u32>(winver.dwBuildNumber));
-    builder.AddData("win-ver-spmajor", static_cast<u32>(winver.wServicePackMajor));
-    builder.AddData("win-ver-spminor", static_cast<u32>(winver.wServicePackMinor));
-  }
+  GetVersionEx(reinterpret_cast<LPOSVERSIONINFO>(&winver));
```

Comment:
I don't think we should be using GetVersionEx because it's been deprecated starting from Windows 8.1. 

>[GetVersionEx may be altered or unavailable for releases after Windows 8.1. Instead, use the Version Helper functions]

https://docs.microsoft.com/en-us/windows/desktop/api/sysinfoapi/nf-sysinfoapi-getversionexa

Reply:
First a less serious answer:
https://twitter.com/Notepad_plus/status/959902540195299331

Now a more serious one:
While it is marked as such, it's definitely not going anywhere anytime soon (how would it work - removed export from the DLL? No chance), and I guess that the "deprecation" is more about the fact Version Helper functions are superior alternatives for almost all use cases of this function - bar Dolphin's one.

Dolphin is also manifested for Windows 10 properly so GetVersionEx will not lie to it, just like RtlGetVersion.

Reply:
Does `RtlGetVersion` also have the returning-the-version-in-the-manifest behavior? Not that there'll be any versions other than Windows 10 soon, but that could be one reason to use it.

The MS-recommended way seems much more awkward: https://docs.microsoft.com/en-us/windows/desktop/SysInfo/getting-the-system-version

Reply:
It does, yes - it was also unaffected by HighVersionLie appverifier option. So indeed, if this is considered a priority this is one reason to stick to the old code - hence my remark that I can recall it from this PR at any time.

Reply:
Ah, then we probably should switch to the file-version method. But I'd be fine with making that a separate PR.

Reply:
You're right, especially since this was not really found using Application Verifier, so it's kinda out of place for this PR. I'll recall this commit and revisit this again at a later point, as I agree file-version method sounds correct - and may even provide info on greater granularity that "Windows version", in an unlikely case of some Windows KB hotfix breaking things.